### DISCUSSION TITLE
error: Tried to reference variable X, but we cannot access variables here

### DISCUSSION THREAD

**Discussion by MartyMcFlyInTheSky** [elkowar/eww#1163] [Q&A] [answered]
I can't reference a variable in deflisten:

```
(defvar pythondir "~/anaconda3/envs/eww/bin/python")

(deflisten wifibox
    :initial "Looking for wifi"
    `${pythondir} ${EWW_CONFIG_DIR}/scripts/dyn_network.py list_networks`
    )
```

Yields an error:

```
myuser@MSR-0317:~$ eww reload
error: Tried to reference variable `pythondir`, but we cannot access variables here
   ┌─ /home/myuser/.config/eww/eww.yuck:24:8
   │
24 │     `${pythondir} ${EWW_CONFIG_DIR}/scripts/dyn_network.py list_networks`
   │        ─────────
   │
   → Expected format: `(deflisten name :initial "0" "tail -f /tmp/example")`
```

What am I doing wrong?

Edit: It also won't work for `defpoll`:

```
(defpoll loading :interval "10ms" "(literal :text ${ loaded ? 'spinny' : 'hook' })")
```

Shows the same error

**Comment by w-lfchen** [comment, marked as answer]
i think it's currently not possible to use the expression language in variable/listener/poll definitions.

parsing that would be pretty difficult i think, just consider recursive definitions

**Reply by MartyMcFlyInTheSky** [reply]
Hmm is there a workaround then?

**Reply by w-lfchen** [reply]
not to my knowledge.
i recommend moving the logic you want to implement into the script you're running.

**Reply by MartyMcFlyInTheSky** [reply]
Thx, did that but it seems a bit hacky tbh.

**Reply by w-lfchen** [reply]
i can't judge the feasability of implementing this.
you could try using `eww get`, that just came to my mind.
